compiler/codegen/symbol.cpp

```diff
@@ -60,16 +60,32 @@
 *                                                                   *
 ********************************* | ********************************/
 
-char llvmFuncDumpName[FUNC_NAME_MAX+1] = "";
-const char *llvmFuncDumpCName = "";
-int llvmFuncOptDump = 0;
+char llvmPrintIrName[FUNC_NAME_MAX+1] = "";
+char llvmPrintIrStage[FUNC_NAME_MAX+1] = "";
+
+int llvmPrintIrNameStageNum;
+const char *llvmPrintIrCName;
+
+std::map<std::string, int> llvmStageMap =
+{
+    {LLVM_NONE_STAGE_NAME, LLVM_NONE_STAGE_NUM},
+    {LLVM_BASIC_STAGE_NAME, LLVM_BASIC_STAGE_NUM},
+    {LLVM_FULL_STAGE_NAME, LLVM_FULL_STAGE_NUM}
+};
+
+std::map<int, std::string> llvmStageRevMap =
```

Comment:
I have to agree with you that map is a bit of overkill. 

In this case it was just easier for me to assign (value)->(key) using std::map notation. Normally I'd like to have std::array or std::vector equivalent looking like this:

```
std::vector llvmStageRevMap =
{
    [LLVM_NONE_STAGE_NUM] = LLVM_NONE_STAGE_NAME,
    [LLVM_BASIC_STAGE_NUM] = LLVM_BASIC_STAGE_NAME,
    [LLVM_FULL_STAGE_NUM] = LLVM_FULL_STAGE_NAME
};
```
But C++ doesn't support given notation.

I consider to even remove map, and simply use bunch of 'if, else' in places where I actually use this: llvmFunctionDump, and verifyStageAndsetStageNum, but this is just my programming style to use map for this kind of task.